chef-config/lib/chef-config/path_helper.rb

```diff
@@ -228,6 +228,17 @@ def self.all_homes(*args)
         joined_paths
       end
     end
+	
+    # On Windows, expands the given path so that it
+    # contains a drive letter at the beginning.
+    # [path] The path to convert
+    def self.get_full_windows_path(path)
+      if ChefConfig.windows?
+        path = cleanpath(File.absolute_path(path))
```

Comment:
This seems unreliable -- it relies on the process current directory, which may differ depending on how Chef is launched. It can also be altered by code (including Chef recipes) that changes the current working directory.

I think for this conversion to work deterministically and to be useful, an unchanging and predictable definition of a path like '/x/y/' on Windows must be defined. Without that, I would actually argue in favor of warnings when such a path is used and eventually (after a major version change) an exception. 

@btm, @jdm, @ksubrama, thoughts? How does `kernel32!CreateFile` behave in this situation?